client/reader/full-post/controller.js

```diff
@@ -81,17 +80,16 @@ export function feedPost( context ) {
 		page.back( context.lastRoute || '/' );
 	}
 
-	renderWithReduxStore(
+	context.primary = (
 		<AsyncLoad
 			require="blocks/reader-full-post"
 			feedId={ feedId }
 			postId={ postId }
 			onClose={ closer }
 			referralStream={ context.lastRoute }
 			onPostNotFound={ renderPostNotFound }
```

Comment:
I'm trying to figure out how this `onPostNotFound` works so that I could decide what to do with [`renderPostNotFound()`](https://github.com/Automattic/wp-calypso/pull/19494/files#diff-c3091eaa05e597c07a7a8971f1e15c46R24).

If I open [non-existing feed post](https://wordpress.com/read/feeds/45339984/posts/123), I don't see _"Post Not Found"_ from `renderPostNotFound()`.

Instead I see _"Post unavailable"_ from [client/blocks/reader-full-post/unavailable.jsx](https://github.com/Automattic/wp-calypso/blob/5059801fe58db65efa27007684f4fa68d7be8e12/client/blocks/reader-full-post/unavailable.jsx#L25), rendered at [blocks/reader-full-post](https://github.com/Automattic/wp-calypso/blob/5059801fe58db65efa27007684f4fa68d7be8e12/client/blocks/reader-full-post/index.jsx#L286-L288).

To me it looks like that's dead code but I might be missing something here?

There's another mention of that prop in [client/blocks/reader-full-post/index.jsx](https://github.com/Automattic/wp-calypso/blob/5059801fe58db65efa27007684f4fa68d7be8e12/client/blocks/reader-full-post/index.jsx#L512) though.

Reply:
I noticed similar issues with `my-sites/domains` so concentrating now to properly test that site selector.

Reply:
Let's ask @Automattic/reader folk. ^^

Reply:
Actually, looks like we don't really have to ask them. A quick `grep` seems to indicate really strongly that this is indeed dead code 😄 Could you file a separate PR to get rid of it?

Reply:
That's what I thought. :-) Will do.